pkg/relayer/cmd/cmd.go

```diff
@@ -134,6 +135,17 @@ func runRelayer(cmd *cobra.Command, _ []string) error {
 		}
 	}
 
+	if relayMinerConfig.Ping.Enabled {
+		ln, err := net.Listen("tcp", relayMinerConfig.Ping.Addr)
+		if err != nil {
+			return fmt.Errorf("failed to listen ping server: %w", err)
+		}
+
+		if err := relayMiner.ServePing(ctx, ln); err != nil {
+			return fmt.Errorf("failed to start ping server: %w", err)
+		}
```

Comment:
I'm sorry, but I'm not sure I understand. Do you want me to extract the `net.Listen(...)` and `relayMiner.ServePing(...)` instructions into another function, such as `func setupPingServer(addr string) error,` in the same package? If yes, I would love to know why as well. Two instructions don't seem too verbose.

Reply:
Thanks for your explanation Bryan, I understand now.

IIRC, I went this way because it was more practical for me to test the `ServePing(...)` when accepting a `net.Listener` interface. The implementation of `net.Listener` could be anything ("TCP" connection on the production code, "Unix domain" connection on the unit test).
But I can manage something to stay consistent and follow that single responsibility principle, yes.